source/Table/Table.js

```diff
@@ -518,10 +519,13 @@ export default class Table extends PureComponent {
 
     if (sortEnabled || onHeaderClick) {
       // If this is a sortable header, clicking it should update the table data's sorting.
-      const newSortDirection =
-        sortBy !== dataKey || sortDirection === SortDirection.DESC
-          ? SortDirection.ASC
-          : SortDirection.DESC;
+      const isFirstTimeSort = sortBy !== dataKey;
+
+      // If this is the firstTime sort of this column, use the column default sort order OR the default, asc.
+      // Otherwise, invert the direction of the sort.
+      const newSortDirection = (isFirstTimeSort ?
+      				(defaultSortOrder || SortDirection.ASC) :
```

Comment:
We could get around this || if we set the default value via `static defaultProps`

Reply:
True. Good point.